test/models/name_test.rb

```diff
@@ -55,7 +55,7 @@ def do_name_parse_test(str, expects)
   end
 
   def assert_name_match_author_required(pattern, string, first_match = string)
-    assert_not(pattern.match(string),
+    assert_not(pattern.to_s.match?(string),
```

Comment:
I don't have time to look into this further tonight, but:

1. If this method is being called with a regexp as the `pattern` argument, I suspect that the assertion won't work correctly if `pattern` is converted to a string.
2. On the extreme, if the method is always called with a regex at the `pattern` argument, then you don't need `to_s` because a regex cannot be `nil` -- `nil` is in its own Ruby class.